includes/compat/blocks/class-blocks-data-extractor.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Class Blocks_Data_Extractor
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay;
+
+use Automattic\WooCommerce\Blocks\Integrations\IntegrationRegistry;
+
+defined( 'ABSPATH' ) || exit; // block direct access.
+
+/**
+ * Extract data fields from certain block based plugins.
+ */
+class Blocks_Data_Extractor {
+
+	/**
+	 * Instance of the integration registry.
+	 *
+	 * @var IntegrationRegistry
+	 */
+	private $integration_registry;
+
+	/**
+	 * An array of blocks to extract data fields.
+	 *
+	 * @var array
+	 */
+	private $blocks = [];
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->integration_registry = new IntegrationRegistry();
+		$this->blocks               = $this->get_available_blocks();
+
+		$this->register_blocks();
+	}
+
+	/**
+	 * Get a list of available Blocks.
+	 */
```

Comment:
For all the docblocks of the methods that return something in this class, it'd probably be good to add the `@return` tag.

Reply:
Added in https://github.com/Automattic/woocommerce-payments/pull/5483/commits/714c44c7c0f0fc1baf64c088a527279929cb10ff